gnd/src/main/java/com/google/android/gnd/repository/OfflineAreaRepository.java

```diff
@@ -85,18 +109,33 @@ private Completable enqueueTileDownloads(OfflineArea area) {
         .andThen(tileDownloadWorkManager.enqueueTileDownloadWorker());
   }
 
+  /**
+   * Determine the set of tiles that need to be downloaded for a given area, then enqueue tile
+   * downloads.
+   */
+  private Completable enqueueTileDownloads(OfflineArea area) {
+    return projectRepository
+        .getActiveProjectOnceAndStream()
+        .compose(Loadable::values)
+        .map(Project::getOfflineBaseMapSources)
+        .map(this::downloadOfflineBaseMapSource)
```

Comment:
This block can be refactored into a new method and then reused in the rest of the code.

Reply:
Good call, thanks! 80e58f8